libethash-cl/CLMiner.cpp

```diff
@@ -432,7 +439,7 @@ void CLMiner::kick_miner()
 {
     // Memory for abort Cannot be static because crashes on macOS.
     const uint32_t one = 1;
-    if (m_abortqueue.size())
+    if (m_abortqueue.size() && !m_settings.noExit)
```

Comment:
The flag should probably be checked first. And also maybe `!m_abortqueue.empty()`?